### ISSUE TITLE
안녕하세요 AXGrid 수정기능 질문이 있어 글 남깁니다.

### ISSUE THREAD

**Issue by fakejolie** [axisj/axisj#702]
AXGrid를 이용하여 게시판을 만들고 있는데, 질문점이 생겨서 글 남깁니다.

회원정보를 수정하고 등록하는 페이지 인데요, modal로 수정창이 뜨면 값을 수정하고 

결과를 반영하게 하고 있습니다. 

그런데..페이징에서 조금 막히는 부분이 생겨서요.

만약 2페이지에서 값을 수정했다면 수정이 완료되고나서 모달창이 닫히고 

그대로 2페이지에 머물러 있어야 그 다음사람의 정보를 수정할 수 있는데

정보를 수정하고 나면 다시 1페이지로 돌아가 버립니다. 

현재 페이징을 해 줄때, 자바단에서 pageNo를 jsonObject에 담아서 

response.setContentType("text/html;charset=utf-8");
		PrintWriter out = response.getWriter();
		out.print(jso);

해주고 있는데요.

      page: {	 paging: true,
			                pageNo: 1,
			                pageSize: 100,
			                status: { formatter: null }
			            }
혹시 이부분때문에 1페이지로 돌아가는건가요?

아니라면 무엇이 잘못되었는지 궁금합니다. 삭제를 했을때도 마찬가지의 문제에 봉착합니다. 

3페이지가 있을 경우 마지막페이지에 게시물이 1개가 있었는데 그 한개가 삭제되면 

맨 첫번째 페이지로 돌아가는 것은 상관없지만 마지막페이지에 게시물 5개중 한개만 삭제한다면 

그 마지막 페이지에 머물러 있어야 하는데 이 역시도 1페이지로 돌아갑니다.

ㅠ_ㅠ 어느 부분을 고쳐야 하는지 감이 안와서요. 

답변 부탁드립니다..ㅠㅠㅠ

**Comment by Happy2You** [comment]
질문만 봐서는 정확히 모르겠습니다..

저도 배우는 입장이라서요 ^^

삭제를 어떤방법으로 하시는지는 모르겠지만..  페이징 값이 잘 안 먹는다고 생각하시면 
해당 pars값을 사용 하시지 마시고, get방식으로 페이지값을 던져서 수동처리 해보시는것도 방법일것 같습니다. 

```
            remove: function () {				

			   var url = "삭제페이지?pageNo=현재 페이지값"
	   
	  		   new AXReq(url, {debug:false, pars:pars, onsucc:function(res){
					   	  if(res.result == AXUtil.ajaxOkCode){;
							  toast.push("삭제가 되었습니다.");
						      }else{
							  toast.push(res.msg);
						      }
					     }
			   });
				   
			   }			   
```

**Comment by hyunjun19** [comment]
http://dev.axisj.com/samples/AXGrid/ajax.html 이 예제의 ajax 통신 내용을 잘 살펴 보세요.
넘어가는 파라미터와 응답으로 받는 데이터 형식을 자세히 보시면 금방 답이 나올껍니다.

ajax 통신내용은 `fiddler`나 크롬의 개발자 도구로 보시면 됩니다.

**Comment by fakejolie** [comment]
음.. json형태로 페이지 값이 파싱되고 있는걸 말씀하시는건가요 ㅠㅠ? 

현재 자바단에서 
  int pageNo=Integer.parseInt(request.getParameter("pageNo"));
		
               jobj.put("pageNo", pageNo);
		jobj.put("pageSize", 100);
		jobj.put("pageCount", pageNum);
		jobj.put("listCount", totalCount);
		JSONObject jso= new JSONObject();
		
		jso.put("result", "ok");
		jso.put("list", jlist);
		jso.put("page", jobj);
		System.out.println(jso);

이런식으로 파라미터값을 넘겨준 다음에 그것을 json 형태로 담아서 
해주고 있는데요. 음.. 페이징 처리는 원활하게 되고있는데, 삭제를 하거나 수정을 했을경우
부모창이 reload가 되면서 1페이지로 돌아가는것 같습니다. ㅠㅠ 이럴경우 스크립트에서 해결해 주나요 ?

**Comment by hyunjun19** [comment]
모달창을 닫고 나서 무슨 처리를 하시나요?
`myGrid.reloadList()`를 호출하면 해당 페이지에 그대로 머물러 있습니다.

**Comment by fakejolie** [comment]
new AXReq(url, {
	            debug: true, pars: pars, onsucc: function (response) {
	            	console.info("res",response.result);
	                if (response.result == AXUtil.ajaxOkCode) {
	                    parent.toast.push("저장되었습니다");
						parent.fnObj.grid.search();
						parent.myModal.close();
							                }else if (response.result == "err") {
	                	parent.toast.push("저장실패");
	                }
	            },
	            onerr: function (response) {
	                AXUtil.alert("오류가 발생했습니다.");
	            }
	        });

	        return false;
	    }

	};
modal.close 밑에 reloadList()를 써야하는건가용?
여기가 업데이트 부분인데요

**Comment by hyunjun19** [comment]
`parent.fnObj.grid.search();` 이 메서드 안에서 페이지를 초기화 하고 있지 않나요?
`parent.fnObj.grid.target.reloadList();` 이렇게 바꿔보세요.

그리고 다음부터는 관련된 부분의 소스와 설명을 상세히 올려주셔야 한 번에 답변을 드릴 수 있습니다. =)

**Comment by fakejolie** [comment]
감사합니다. 수정부분은 해결을 했는데요.. 삭제부분에서 또 똑같은 이유로 막히네요 ㅠㅠ 

			remove: function(){
					var checkedList = myGrid.getCheckedList(0);
					console.info("checked",checkedList)
					if(checkedList.length == 0){
						alert("선택된 목록이 없습니다. 삭제하시려는 목록을 체크하세요");	
						return;
					}
					if(!confirm("정말 삭제 하시겠습니까?")) return;
					//var id=checkedList[0].id;
					mask.open();
					//var param="?id="+id;
					var url = "/adminAuth/delete";
					 var pas = [];
					$.each(checkedList, function(){
						pas.push("auth_id="+this.auth_id);
					}); 
					var pars = pas.join("&")
					console.info("pars",pars)
					new AXReq(url, {debug:false, pars:pars, 
						onsucc:function(response){
							if(response.result == AXUtil.ajaxOkCode){
								toast.push("삭제 되었습니다.");
								fnObj.grid.target.reloadList();
							}else{
								AXUtil.alert(response);
							}
							mask.close();
						},
						onerr:null
					});	
				}   
			}
	    };

myGrid.reloadList()도 써 보았는데 ㅠㅠ 여기서는 1페이지로 돌아가 버리네요

**Comment by hyunjun19** [comment]
소스 코드상으로는 별 문제가 없어 보입니다.
해당 페이지를 직접 봐야 원인을 알것 같습니다.
